TextSelection/Views/MultitouchTextView.swift

```diff
@@ -0,0 +1,225 @@
+//
+//  MultitouchTextView.swift
+//  TextSelection
+//
+//  Created by Lacy Rhoades on 5/15/16.
+//  Copyright © 2016 Ink & Switch. All rights reserved.
+//
+
+import UIKit
+
+class MultitouchTextView: UITextView {
+    private var highlightSelection: Selection?
+    private var preSelection: Selection?
+    private var selection: Selection?
+    
+    var customDelegate: MultitouchTextViewDelegate?
+    
+    override init(frame: CGRect, textContainer: NSTextContainer?) {
+        super.init(frame: frame, textContainer: textContainer)
+        
+        self.setup()
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        
+        self.setup()
+    }
+    
+    func setup() {
+        self.editable = false
+        
+        self.gestureRecognizers?.forEach({ (recognizer) in
+            switch Mirror(reflecting: recognizer).subjectType {
+            case is UILongPressGestureRecognizer.Type:
+                // For normal selections
+                self.removeGestureRecognizer(recognizer)
+                break
+            case is UITapGestureRecognizer.Type:
+                // Not sure what this is
+                // Probably "cancels" selections
+                self.removeGestureRecognizer(recognizer)
+                break
+            case is UIPanGestureRecognizer.Type:
+                // For scrolling
+                let pan = recognizer as! UIPanGestureRecognizer
+                pan.minimumNumberOfTouches = 2
+                break
+            default:
+                let className = String(Mirror(reflecting: recognizer).subjectType)
+                if (className == "UITapAndAHalfRecognizer") {
+                    // For tap and drag
+                    self.removeGestureRecognizer(recognizer)
+                } else if (className == "_UIPreviewInteractionTouchObservingGestureRecognizer") {
+                    // Not sure what this is
+                }
+                break
+            }
+        })
+    }
+    
+    var baseAttributes: [String: AnyObject] {
+        get {
+            return [NSForegroundColorAttributeName: UIColor.darkTextColor()]
+        }
+    }
+    
+    var preSelectionTimer: NSTimer?
+    
+    func clearPreSelectionWithDelay() {
+        self.preSelectionTimer?.invalidate()
+        self.preSelectionTimer = NSTimer.scheduledTimerWithTimeInterval(2.0, target: self, selector: #selector(clearPreSelection), userInfo: nil, repeats: false)
+    }
+    
+    func clearPreSelection() {
+        self.preSelection = nil
+        self.updateViews(true)
+        self.selectionDidChange()
+    }
+    
+    func didTapButton() {
+        if (self.selection != nil && self.preSelection == nil) {
+            self.clearSelection()
+        } else {
+            self.appendPreSelectionToSelection()
+        }
+    }
+    
+    var shouldShowControlButton: Bool {
+        get {
+            return self.selection != nil || self.preSelection != nil
```

Comment:
Moved the "show/hide big blue button" logic inside this class. No one else cares about the specifics.